### PR TITLE
Add doc explaining remote parts

### PR DESCRIPTION
**Pull request by popey**
Added first pass at a doc explaining how to consume and create remote parts. Based in part off a 2 year old blog post from Sergio.

### CODE REVIEW

File: build-snaps/remote-parts.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
+---
+layout: base
+title: Remote parts
+---
+
+Developers using snapcraft to build snaps can leverage pre-existing 'remote parts' in their snap, to save time re-implementing existing components.
+
+Developers who have created local parts which may be of use to others, can submit them to the remote part cache for other snapcrafter consume.
```
Comment (caldav) [inline build-snaps/remote-parts.md]: snapcrafter -> developers?
**to** consume

File: build-snaps/remote-parts.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+Developers using snapcraft to build snaps can leverage pre-existing 'remote parts' in their snap, to save time re-implementing existing components.
+
+Developers who have created local parts which may be of use to others, can submit them to the remote part cache for other snapcrafter consume. 
+
+## Consuming remote parts
+
+### Search for parts
+
+The Snapcraft command line tool can search the remote parts cache using the `search` command. Before issuing `search` use the `update` to update the parts list from the remote cache.
```
Comment (caldav) [inline build-snaps/remote-parts.md]: before issuing `search`, use the `update` command

File: build-snaps/remote-parts.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+```
+$ snapcraft update
+Downloading parts list
+```
+
+Once updated, `search` on its own will return the entire list of remote parts, or specify a search term to restrict the results returned.
```
Comment (caldav) [inline build-snaps/remote-parts.md]: , you can specify a search term

File: build-snaps/remote-parts.md [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+parts:
+    client:
+       plugin: autotools
+       source: .
+       after: [curl]
+```
+
+#### Composing
+
+Perhaps the remote part is almost what's needed, but a change is required to fit the needs of your snap. In this case we can override pieces of the remote part. In this example we're overriding the source URL.
```
Comment (caldav) [inline build-snaps/remote-parts.md]: Perhaps a remote part is almost the one you need
required to fit your snap

File: build-snaps/remote-parts.md [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+        source-type: tar
+```
+
+## Creating remote parts
+
+If you've created a part which might be useful for other developers, it's possible to share them easily.
+
+### Host part
+
+Create a repo containing only the part you wish to share. For example [https://github.com/sergiusens/curl](https://github.com/sergiusens/curl)contains the curl part mentioned above.
```
Comment (caldav) [inline build-snaps/remote-parts.md]: Space missing between link and "contains"

File: build-snaps/remote-parts.md [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+
+### Update parts wiki
+
+Add a yaml formatted entry to the [parts wiki](https://wiki.ubuntu.com/snapcraft/parts) page.
```
Comment (caldav) [inline build-snaps/remote-parts.md]: YAML

File: build-snaps/remote-parts.md [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+maintainer: Sergio Schvezov <sergio.schvezov@ubuntu.com>
+description:
+  A tool and a library (usable from many languages) for
+  client side URL transfers, supporting FTP, FTPS, HTTP,
+  HTTPS, TELNET, DICT, FILE and LDAP.
+parts: [curl]
+---
+```
+
+*Note:* To edit the Ubuntu wiki you'll need an [Ubuntu SSO](https://login.ubuntu.com/) account (as used in the [snap store](https://wiki.ubuntu.com/snapcraft/parts)), and need to request to join the [ubuntu-wiki-editors](https://launchpad.net/~ubuntu-wiki-editors) team. Once approved, logout from the Ubuntu wiki and log back in again to refresh your new credentials.
```
Comment (caldav) [inline build-snaps/remote-parts.md]: Not sure the store link is the right one here

**Comment by popey** [comment]
Thanks for the review, updated!

**Pull request merged**